app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -359,6 +377,24 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         BrowsersCache.resetAll()
     }
 
+    private fun getBookmarkCount(node: BookmarkNode): Int {
+        if (node.children == null) {
+            return 0
+        }
+
+        var count = 0
+
+        for (child in node.children!!) {
```

Comment:
I've always been told in reviews not to use the '!!', so just wondering why it's used here and if there's another way this could be done.

Reply:
Agreed. This should be in an `else`, since line 381 should be filtering out anything that doesn't have children so we don't need the `!!`

Reply:
<img width="936" alt="Screen Shot 2020-12-11 at 3 44 16 PM" src="https://user-images.githubusercontent.com/1190888/101953105-c4219d00-3bc7-11eb-9e87-8040f9028a70.png">

Android studio complained without the !! so I didn't have anyway around this even with the if/else

Reply:
```
private fun getBookmarkCount(node: BookmarkNode): Int {
        if (node.children == null) {
            return 0
        } else {
          var count = 0
          for (child in node.children) {
              if (child.type == BookmarkNodeType.FOLDER) {
                  count += getBookmarkCount(child)
              } else if (child.type == BookmarkNodeType.ITEM) {
                  count++
              }
           }
           return count
        }
    }
```